src/scheduler/job_info.c

```diff
@@ -2901,11 +2901,9 @@ find_and_preempt_jobs(status *policy, int pbs_sd, resource_resv *hjob, server_in
 				sinfo->num_preempted++;
 				/* We need to correct our calendar */
 				if (sinfo->calendar != NULL) {
-					te = find_timed_event(sinfo->calendar->events, job->name, TIMED_END_EVENT, 0);
-					if (te != NULL) {
-						if (delete_event(sinfo, te, DE_NO_FLAGS) == 0)
-							schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_JOB, LOG_INFO, job->name, "Failed to delete end event for job.");
-					}
+					te = find_timed_event(sinfo->calendar->events, 0, job->name, TIMED_END_EVENT, 0);
```

Comment:
Didn't we decide on moving this to update_resresv_on_end() function. That is a better place for it.

Reply:
I was of the understanding that we need not move the deletion of event as  is being called from calendaring code as well and just disable the event.
Did you mean to disable in update_resresv_on_end()?

Reply:
Yes I meant move disable event being moved to update_resresv_on_end()

Reply:
Ok. Moved the disabling of event to update_resresv_on_end().